### ISSUE TITLE
Generated module fails to build due to circular reference error and namespace conflict.

### ISSUE THREAD

**Issue by peombwa** [Azure/autorest.powershell#517]
After generating PowerShell module using [users.drive.yml](https://gist.github.com/peombwa/75d04e3d4b57a6d8bfbda0929aee92c4#file-failing-user-drive-yml) file, building the module results in the following build errors:
``` ps
generated\cmdlets\NewUserDrive_CreateExpanded.cs(9,42): error CS0120: An object reference is required for the non-static field, method, or property 'NewUserDrive_CreateExpanded.System' [E:\MicrosoftGraph\msgraph-sdk-powershell\src\Users.Drive\Users.Drive\Graph.Users.Drive.csproj]
generated\cmdlets\NewUserDrive_CreateViaIdentityExpanded.cs(9,42): error CS0120: An object reference is required for the non-static field, method, or property 'NewUserDrive_CreateViaIdentityExpanded.System' [E:\MicrosoftGraph\msgraph-sdk-powershell\src\Users.Drive\Users.Drive\Graph.Users.Drive.csproj]
generated\cmdlets\UpdateUserDrive_UpdateExpanded.cs(9,42): error CS0120: An object reference is required for the non-static field, method, or property 'UpdateUserDrive_UpdateExpanded.System' [E:\MicrosoftGraph\msgraph-sdk-powershell\src\Users.Drive\Users.Drive\Graph.Users.Drive.csproj]
generated\cmdlets\UpdateUserDrive_UpdateExpanded1.cs(9,42): error CS0120: An object reference is required for the non-static field, method, or property 'UpdateUserDrive_UpdateExpanded1.System' [E:\MicrosoftGraph\msgraph-sdk-powershell\src\Users.Drive\Users.Drive\Graph.Users.Drive.csproj]
generated\cmdlets\UpdateUserDrive_UpdateViaIdentityExpanded1.cs(9,42): error CS0120: An object reference is required for the non-static field, method, or property 'UpdateUserDrive_UpdateViaIdentityExpanded1.System' [E:\MicrosoftGraph\msgraph-sdk-powershell\src\Users.Drive\Users.Drive\Graph.Users.Drive.csproj]
generated\cmdlets\UpdateUserDrive_UpdateViaIdentityExpanded.cs(9,42): error CS0120: An object reference is required for the non-static field, method, or property 'UpdateUserDrive_UpdateViaIdentityExpanded.System' [E:\MicrosoftGraph\msgraph-sdk-powershell\src\Users.Drive\Users.Drive\Graph.Users.Drive.csproj]
generated\api\Models\MicrosoftGraphListItem.cs(6,9): error CS0535: 'MicrosoftGraphListItem' does not implement interface member 'IComponentsSchemasMicrosoftGraphBaseitemAllof1.CreatedBy' [E:\MicrosoftGraph\msgraph-sdk-powershell\src\Users.Drive\Users.Drive\Graph.Users.Drive.csproj]
generated\api\Models\MicrosoftGraphListItem.cs(6,9): error CS0535: 'MicrosoftGraphListItem' does not implement interface member 'IComponentsSchemasMicrosoftGraphBaseitemAllof1.CreatedByUserDeletedDateTime' [E:\MicrosoftGraph\msgraph-sdk-powershell\src\Users.Drive\Users.Drive\Graph.Users.Drive.csproj]
generated\api\Models\MicrosoftGraphListItem.cs(6,9): error CS0535: 'MicrosoftGraphListItem' does not implement interface member 'IComponentsSchemasMicrosoftGraphBaseitemAllof1.CreatedByUserDrive' [E:\MicrosoftGraph\msgraph-sdk-powershell\src\Users.Drive\Users.Drive\Graph.Users.Drive.csproj]
generated\api\Models\MicrosoftGraphListItem.cs(6,9): error CS0535: 'MicrosoftGraphListItem' does not implement interface member 'IComponentsSchemasMicrosoftGraphBaseitemAllof1.CreatedByUserDrives' [E:\MicrosoftGraph\msgraph-sdk-powershell\src\Users.Drive\Users.Drive\Graph.Users.Drive.csproj]
generated\api\Models\MicrosoftGraphListItem.cs(6,9): error CS0535: 'MicrosoftGraphListItem' does not implement interface member 'IComponentsSchemasMicrosoftGraphBaseitemAllof1.CreatedByUserId' [E:\MicrosoftGraph\msgraph-sdk-powershell\src\Users.Drive\Users.Drive\Graph.Users.Drive.csproj]
generated\api\Models\MicrosoftGraphListItem.cs(6,9): error CS0535: 'MicrosoftGraphListItem' does not implement interface member 'IComponentsSchemasMicrosoftGraphBaseitemAllof1.CreatedByUserJoinedTeams' [E:\MicrosoftGraph\msgraph-sdk-powershell\src\Users.Drive\Users.Drive\Graph.Users.Drive.csproj]
generated\api\Models\MicrosoftGraphListItem.cs(6,9): error CS0535: 'MicrosoftGraphListItem' does not implement interface member 'IComponentsSchemasMicrosoftGraphBaseitemAllof1.CreatedByUserOnenoteId' [E:\MicrosoftGraph\msgraph-sdk-powershell\src\Users.Drive\Users.Drive\Graph.Users.Drive.csproj]
generated\api\Models\MicrosoftGraphListItem.cs(6,9): error CS0535: 'MicrosoftGraphListItem' does not implement interface member 'IComponentsSchemasMicrosoftGraphBaseitemAllof1.CreatedByUserOnenoteNotebooks' [E:\MicrosoftGraph\msgraph-sdk-powershell\src\Users.Drive\Users.Drive\Graph.Users.Drive.csproj]
generated\api\Models\MicrosoftGraphListItem.cs(6,9): error CS0535: 'MicrosoftGraphListItem' does not implement interface member 'IComponentsSchemasMicrosoftGraphBaseitemAllof1.CreatedByUserOnenotePages' [E:\MicrosoftGraph\msgraph-sdk-powershell\src\Users.Drive\Users.Drive\Graph.Users.Drive.csproj]
generated\api\Models\MicrosoftGraphListItem.cs(6,9): error CS0535: 'MicrosoftGraphListItem' does not implement interface member 'IComponentsSchemasMicrosoftGraphBaseitemAllof1.CreatedByUserOnenoteSectionGroups' [E:\MicrosoftGraph\msgraph-sdk-powershell\src\Users.Drive\Users.Drive\Graph.Users.Drive.csproj]
generated\api\Models\MicrosoftGraphListItem.cs(6,9): error CS0535: 'MicrosoftGraphListItem' does not implement interface member 'IComponentsSchemasMicrosoftGraphBaseitemAllof1.CreatedByUserOnenoteSections' [E:\MicrosoftGraph\msgraph-sdk-powershell\src\Users.Drive\Users.Drive\Graph.Users.Drive.csproj]
...
...
...
```

During generation, the following warning messages are logged:
``` ps
Note: model microsoft.graph.sectionGroup has a circular reference, and we're skipping inlining.
  microsoft.graph.onenoteSection => components·schemas·microsoft-graph-onenotesection·allof·1::parentSectionGroup => microsoft.graph.sectionGroup => components·schemas·microsoft-graph-sectiongroup·allof·1::parentSectionGroup
Note: model microsoft.graph.baseItem has a circular reference, and we're skipping inlining.
  microsoft.graph.site => microsoft.graph.baseItem => components·schemas·microsoft-graph-baseitem·allof·1::createdByUser => microsoft.graph.user => components·schemas·microsoft-graph-user·allof·1::drive => microsoft.graph.drive
Note: model microsoft.graph.drive has a circular reference, and we're skipping inlining.
  microsoft.graph.site => microsoft.graph.baseItem => components·schemas·microsoft-graph-baseitem·allof·1::createdByUser => microsoft.graph.user => components·schemas·microsoft-graph-user·allof·1::drive => microsoft.graph.drive => components·schemas·microsoft-graph-drive·allof·1::list => microsoft.graph.list => components·schemas·microsoft-graph-list·allof·1::drive
Note: model microsoft.graph.driveItem has a circular reference, and we're skipping inlining.
  microsoft.graph.site => microsoft.graph.baseItem => components·schemas·microsoft-graph-baseitem·allof·1::createdByUser => microsoft.graph.user => components·schemas·microsoft-graph-user·allof·1::drive => microsoft.graph.drive => components·schemas·microsoft-graph-drive·allof·1::root => microsoft.graph.driveItem => components·schemas·microsoft-graph-driveitem·allof·1::listItem => microsoft.graph.listItem => components·schemas·microsoft-graph-listitem·allof·1::driveItem
Note: model microsoft.graph.post has a circular reference, and we're skipping inlining.
  microsoft.graph.post => components·schemas·microsoft-graph-post·allof·1::inReplyTo
```
From the build error logs, it looks two issues are present:

1. [Users.Drive.yml](https://gist.github.com/peombwa/75d04e3d4b57a6d8bfbda0929aee92c4#file-failing-user-drive-yml) has a property named `system` on `#components/schemas/microsoft.graph.drive` model. This conflicts with CSharp's `System` namespace in the affected files. Maybe you can prefix all CSharp's `System.XYZ` with `global::System.XYZ`. In this case, `global::System.Management.Automation`.

2. Is there a way to gracefully break/skip circular referenced models without causing above build errors. Maybe something like a directive that can we can use to specify a maximum circular reference depth and when to break it.

## Repro Steps
`autorest-beta --title:"Users.Drive" autoRestConfig.yml --verbose`

**autoRestConfig.yml**
``` yml
powershell: true
version: latest
use: '@autorest/powershell@latest'
module-version: 0.1.0
subject-prefix: ''
azure: false
namespace: Microsoft.Graph.PowerShell
input-file: users.drive.yml
output-folder: ..\src\$(title)\$(title)
clear-output-folder: true
use-internal-constructors: true
use-datetimeoffset : true
directive:
  - no-inline:
    - microsoft.graph.sharepointIds
    - microsoft.graph.identitySet
    - microsoft.graph.itemReference
    - microsoft.graph.directoryObject
```
- [users.drive.yml](https://gist.github.com/peombwa/75d04e3d4b57a6d8bfbda0929aee92c4#file-failing-user-drive-yml)
- autorest-beta@3.0.6121
- @autorest/powershell@2.0.309
- @autorest/remodeler@2.0.8

**Comment by darrelmiller** [comment]
hey @fearthecowboy Any suggestions on how we might work around this issue?  It is one final issue that is breaking quite a few of the modules that we are trying generate and preventing us from sharing a preview of the modules.

**Comment by fearthecowboy** [comment]
O_O 

I thought we had all the namespaces prefixed with `global:: ` .. I've found those and fixed them, I'll have it thru CI soon.

I've removed the inlining message, it was just cosmetic anyway.

I've tweaked the remodeler to use `title` as a typename with inlined models. Never seen that before, but it makes it cleaner (gets rid of `....AllOf1` crap)

I'm looking at why it's buggering up implementing the parent interfaces right now.
